docs/content/concepts/graphql-indexer.mdx

```diff
@@ -0,0 +1,367 @@
+- Needs to display structured results in a frontend (e.g., wallets, dashboards)
+- Benefits from flexible, composable queries that reduce overfetching
+- Relies on multiple data entities (e.g., transactions + objects + events) in a single request, or in a consistent fashion when spread over multiple requests (as if the responses came from a snapshot at some checkpoint).
+
+## Deployment options
+
+You can run or use the GraphQL and Indexer data stack in the following configurations:
+
+### Fully managed service
+
+As a developer, you can access GraphQL as a service from an indexer operator or data provider who runs and operates the full stack behind the scenes. Reach out to your data provider and ask if they already offer or plan to offer this service.
+
+### Partial self-managed
+
+As a developer, you can:
+
+- Run the Indexer pipelines and GraphQL service, while using the Archival Service from an indexer operator or data provider, and a full node from an RPC provider.
+- Configure and manage a Postgres-compatible database (local Postgres, AlloyDB, and so on) as the primary data store.
+- Deploy the self-managed components on cloud infrastructure or baremetal.
+
+:::info
+
+You cannot run and operate the GraphQL service in self-managed configurations just yet. The functionality will become available after required changes are made to how the GraphQL service integrates with the Archival Service.
+
+:::
+
+### Fully self-managed
+
+As a developer, indexer operator, or data provider, you can:
+
+- Run the complete stack: Indexer pipelines, GraphQL service, Postgres-compatible database, Archival Service, Consistent Store and full node on cloud infrastructure or bare metal.
+- Serve GraphQL to your own applications or to other builders and third-party services.
+
+## Working with the GraphQL service
+
+The GraphQL service exposes a query surface conforming to [GraphQL concepts](/concepts/graphql-rpc). It allows pagination, filtering, and consistent snapshot queries. The service also supports runtime configuration for schema, query cost limits, and logging.
+
+The GraphQL schema is defined in the [GraphQL reference](/references/sui-graphql). You can explore supported types and fields there, use the GraphiQL IDE to test queries, and read documentation on the up-to-date schema.
+
+GraphQL is deployed as a single binary implementing a stateless, horizontally scalable service. Queries are served with data from one or more of a Postgres-compatible database (filters over historical data), Archival Service (point lookups), Consistent Store (live data), or full node (execution and simulation), based on need. Access to these stores must be configured with the service on start-up, otherwise the service might fail to respond correctly to requests. More details on how to set-up, configure, and run the service is available in its [README](https://github.com/MystenLabs/sui/tree/main/crates/sui-indexer-alt-graphql).
```

Comment:
```suggestion
GraphQL service is deployed as a single binary implementing a stateless, horizontally scalable service. Queries are served with data from one or more of a Postgres-compatible database (filters over historical data), Archival Service (point lookups), Consistent Store (live data), or full node (execution and simulation), based on need. Access to these stores must be configured with the service on start-up, otherwise the service might fail to respond correctly to requests. More details on how to set-up, configure, and run the service is available in its [README](https://github.com/MystenLabs/sui/tree/main/crates/sui-indexer-alt-graphql).
```